src/cli-validator/runValidator.js

```diff
@@ -230,13 +231,21 @@ const processInput = async function(program) {
       exitCode = 1;
       continue;
     }
-
+    if (errorsOnly) {
+      results.warning = false;
+    }
     if (jsonOutput) {
-      printJson(results, originalFile);
+      printJson(results, originalFile, errorsOnly);
     } else {
       if (results.error || results.warning) {
-        print(results, chalk, printValidators, reportingStats, originalFile);
-        // fail on errors, but not if there are only warnings
```

Comment:
I don't want to lose this comment, it should be right above `if (results.error) exitCode = 1;`